packages/txservice/src/dispatch.ts

```diff
@@ -314,105 +315,123 @@ export class TransactionDispatch extends ChainRpcProvider {
       chainId: this.chainId,
-      minTx: { to: minTx.to, from: minTx.from },
       txsId,
     });
 
-    const result = await this.queue.add(async (): Promise<{ value: Transaction | NxtpError; success: boolean }> => {
-      try {
-        // Wait until there's room in the buffer.
-        if (this.inflightBuffer.isFull) {
-          this.logger.warn("Inflight buffer is full! Waiting in queue to send.", requestContext, methodContext, {
-            chainId: this.chainId,
-            bufferLength: this.inflightBuffer.length,
-            txsId,
-          });
-          while (this.inflightBuffer.isFull) {
-            await delay(200);
+    const result = await this.queue.add(
+      async (): Promise<{ value: OnchainTransaction | NxtpError; success: boolean }> => {
+        try {
+          // Wait until there's room in the buffer.
+          if (this.inflightBuffer.isFull) {
+            this.logger.warn("Inflight buffer is full! Waiting in queue to send.", requestContext, methodContext, {
+              chainId: this.chainId,
+              bufferLength: this.inflightBuffer.length,
+              txsId,
+            });
+            while (this.inflightBuffer.isFull) {
+              // TODO: This delay was raised to help alleviate a "trickling bottleneck" when the inflight buffer remains full for
+              // an extended period. An alternative: maybe we should wait until the buffer falls *below* a certain threshold?
+              await delay(10_000);
+            }
           }
-        }
 
-        // Estimate gas here will throw if the transaction is going to revert on-chain for "legit" reasons. This means
-        // that, if we get past this method, we can *generally* assume that the transaction will go through on submit - although it's
-        // still possible to revert due to a state change below.
-        let gas = await this.getGas(minTx, requestContext);
-        if (this.chainId === 42161) {
-          gas = new Gas(gas.baseValue, BigNumber.from(10_000_000));
-        }
+          // Estimate gas here will throw if the transaction is going to revert on-chain for "legit" reasons. This means
+          // that, if we get past this method, we can *generally* assume that the transaction will go through on submit - although it's
+          // still possible to revert due to a state change below.
+          const attemptedNonces: number[] = [];
+          const [gasLimit, gasPrice, nonceInfo] = await Promise.all([
+            this.estimateGas(minTx),
+            this.getGasPrice(requestContext),
+            this.determineNonce(attemptedNonces),
+          ]);
+          let { nonce, backfill, transactionCount } = nonceInfo;
+
+          // TODO: Remove hardcoded (exposed gasLimitInflation config var should replace this).
+          let gas = new Gas(gasPrice, gasLimit);
+          if (this.chainId === 42161) {
+            gas = new Gas(gas.baseValue, BigNumber.from(10_000_000));
+          }
```

Comment:
oof wow dont love the hardcoded chain id exceptions, but totally understand why they happen. adding a comment here would be good (i already can't remember, and i know this isn't from this PR)

Reply:
Yeah but it looks like it might be outdated? not sure I will look into this

Reply:
nope not outdated, same thing as it was previously lol